File: cogs/general.py
import discord
from discord.ext import commands

class General(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.command(name="ping")
    async def ping(self, ctx):
        await ctx.send(f"Pong! {round(self.bot.latency * 1000)}ms")

    @commands.command(name="serverinfo")
    async def serverinfo(self, ctx):
        guild = ctx.guild
        embed = discord.Embed(title=f"Server Info - {guild.name}", color=discord.Color.green())
        embed.set_thumbnail(url=guild.icon.url if guild.icon else None)
        embed.add_field(name="Owner", value=guild.owner.mention)
        embed.add_field(name="Members", value=guild.member_count)
        embed.add_field(name="Roles", value=len(guild.roles))
        embed.add_field(name="Created At", value=guild.created_at.strftime("%Y-%m-%d"))
        await ctx.send(embed=embed)

    @commands.command(name="userinfo")
    async def userinfo(self, ctx, member: discord.Member = None):
        member = member or ctx.author
        embed = discord.Embed(title=f"User Info - {member.name}", color=member.color)
        embed.set_thumbnail(url=member.display_avatar.url)
        embed.add_field(name="ID", value=member.id)
        embed.add_field(name="Joined Server", value=member.joined_at.strftime("%Y-%m-%d"))
        embed.add_field(name="Joined Discord", value=member.created_at.strftime("%Y-%m-%d"))
        roles = [role.mention for role in member.roles if role != ctx.guild.default_role]
        embed.add_field(name=f"Roles ({len(roles)})", value=" ".join(roles) if roles else "No roles")
        await ctx.send(embed=embed)

async def setup(bot):
    await bot.add_cog(General(bot))

File: src/commands/general.js
const { SlashCommandBuilder, EmbedBuilder } = require('discord.js');

module.exports = {
	data: new SlashCommandBuilder()
		.setName('info')
		.setDescription('General information commands')
		.addSubcommand(subcommand => subcommand.setName('ping').setDescription('Check bot latency'))
		.addSubcommand(subcommand => subcommand.setName('server').setDescription('Get server info'))
		.addSubcommand(subcommand =>
			subcommand
				.setName('user')
				.setDescription('Get user info')
				.addUserOption(option => option.setName('target').setDescription('The user to get info for')))
		.addSubcommand(subcommand =>
			subcommand
				.setName('avatar')
				.setDescription('Get user avatar')
				.addUserOption(option => option.setName('target').setDescription('The user to get avatar for'))),
	async execute(interaction) {
		const subcommand = interaction.options.getSubcommand();
		const embed = new EmbedBuilder().setColor('Blue').setTimestamp();

		if (subcommand === 'ping') {
			return interaction.reply(`Pong! Latency: ${interaction.client.ws.ping}ms`);
		}

		if (subcommand === 'server') {
			const { guild } = interaction;
			embed.setTitle(`Server Info - ${guild.name}`)
				.setThumbnail(guild.iconURL())
				.addFields(
					{ name: 'Owner', value: `<@${guild.ownerId}>`, inline: true },
					{ name: 'Members', value: `${guild.memberCount}`, inline: true },
					{ name: 'Created At', value: `${guild.createdAt.toDateString()}`, inline: true },
				);
			return interaction.reply({ embeds: [embed] });
		}

		if (subcommand === 'user') {
			const target = interaction.options.getUser('target') || interaction.user;
			const member = await interaction.guild.members.fetch(target.id);
			embed.setTitle(`User Info - ${target.tag}`)
				.setThumbnail(target.displayAvatarURL())
				.addFields(
					{ name: 'ID', value: target.id, inline: true },
					{ name: 'Joined Server', value: member.joinedAt.toDateString(), inline: true },
					{ name: 'Joined Discord', value: target.createdAt.toDateString(), inline: true },
				);
			return interaction.reply({ embeds: [embed] });
		}

		if (subcommand === 'avatar') {
			const target = interaction.options.getUser('target') || interaction.user;
			embed.setTitle(`${target.tag}'s Avatar`).setImage(target.displayAvatarURL({ size: 1024 }));
			return interaction.reply({ embeds: [embed] });
		}
	},
};
